### PR TITLE
Feature captions performance

### PR DESCRIPTION
**Pull request by crazypsycho**
Improves performance
Add captions to elementor widget
Add some more options
Add image-source from elementor dynamic-tag (like acf-gallery)
Fix issues

### CODE REVIEW

File: trunk/lib/PostGalleryImageList.php [hunk 1]
```diff
@@ -355,4 +413,27 @@ public static function getFirstImage( $size = 'post-thumbnail', $post_id = null
             'size' => $size,
         ];
     }
+
+
+    public static function getAllAttachmentIds() {
```
Comment (eac0n) [inline trunk/lib/PostGalleryImageList.php]: According to its behavior this is not a getter method. The method should either return all attachment ids or be renamed.
PhpDocs should be added,

File: trunk/public/css/post-gallery-public.css [hunk 2]
```diff
@@ -214,6 +218,37 @@ body.liteboxgallery-open {
...
+  position: absolute;
+  font-size: 1rem;
+  pointer-events: none;
+  left: 0;
+  right: 0;
+  width: auto;
+  transition: all 0.3s ease;
+  z-index: 100;
+}
+/*
```
Comment (eac0n) [inline trunk/public/css/post-gallery-public.css]: Unused code should be removed.

File: trunk/public/PostGalleryPublic.php [hunk 3]
```diff
@@ -445,6 +453,47 @@ public function returnGalleryHtml( $template = '', $postid = 0, $args = [] ) {
...
+                $appendList[$item['position_to_append']] = [];
+            }
+
+            $appendList[$item['position_to_append']][] = $item['template_to_append'];
+        }
+
+        return $appendList;
+    }
+
+    public function getCaption( $image ) {
```
Comment (eac0n) [inline trunk/public/PostGalleryPublic.php:486]: PhpDocs are missing.
Method should return a default value.

File: trunk/templates/thumbs.php [hunk 4]
```diff
@@ -7,10 +7,21 @@
...
  */
 ?>
-    <figure
+    <figure role="group"
             class="gallery pg-theme-thumbs pg-theme-list <?php echo $this->option( 'containerClass' ); ?>">
         <?php foreach ( $images as $image ): ?>
             <?php
+            $count += 1;
+
+            if ( !empty( $appendList[$count] ) ) {
```
Comment (eac0n) [inline trunk/templates/thumbs.php]: In template files the alternative syntax for control structures should be used.

**Review by eac0n — COMMENTED** [review]
`dev` should be the base of a feature branch.

File: trunk/lib/PostGalleryImageList.php (hunk 1 above)
Comment (crazypsycho) [reply trunk/lib/PostGalleryImageList.php]: Renamed to loadAllAttachmentIds

File: trunk/lib/PostGalleryImageList.php [hunk 5]
```diff
@@ -94,60 +100,112 @@ public static function get( int $postid = null ): array {
...
         return $images;
     }
 
+    /**
+     * Get list with attachments-data from an attachment-list
+     *
+     * @param array $imageIdList
+     * @return array
+     */
+    public static function getByDynamic( $imageIdList ): array {
```
Comment (eac0n) [inline trunk/lib/PostGalleryImageList.php]: Rethink namings.
What is a "Dynamic"? A `$imageIdList` is expected so `getByImageIdList` would be obvious.
The PhpDocs are telling something about attachment-data and attachments-list.
Please rename the everything according to what this method actualy does?

**Review by eac0n — CHANGES_REQUESTED** [review]

File: trunk/lib/PostGalleryImageList.php (hunk 5 above)
Comment (crazypsycho) [reply trunk/lib/PostGalleryImageList.php]: Renamed to getByAttachmentIds( $attachmentIds )

**Review by eac0n — APPROVED** [review]

**Pull request merged**